build_tools/error_filter.py

```diff
@@ -64,8 +64,12 @@ def parse_error(self, line):
 
 class CompilerErrorParser(MatchErrorParser):
     def __init__(self):
-        # format: '<filename>:<line #>:<column #>: error: <error msg>'
-        super(CompilerErrorParser, self).__init__(r'\S+:\d+:\d+: error:')
+        # format (compile error):
+        #   '<filename>:<line #>:<column #>: error: <error msg>'
+        # format (compile error):
+        #   '<filename>:<line #>: error: <error msg>'
+        # The below regex catches both
```

Comment:
In the case of compile error, <line #>: will match \S+: and <column #>: will match \d+. In case of link error, <filename #>: will match \S+: and <line #>: will match \d+:.